metabase/metabase
# Add warning message when trying to navigate away from editing a model's query or metadata

### Description

Adds a `beforeunload` event when the user tries to edit a model that they have already created. If the user makes any changes to the model's query or metadata, then tries to refresh, close, or navigate to a new URL, the browser should popup with a "Are you sure you want to leave?" message that the user must interact with before continuing on.

### How to verify

1. Create and save a new model.
2. Try to edit the model's query or metadata.
3. Try to refresh the page, close the tab, or go to a different URL.
4. The browser should pop up with a message similar to the screenshot below.

### Demo

![Screenshot 2023-04-21 at 4 34 47 PM](https://user-images.githubusercontent.com/25306947/233649374-9f80c4b4-1c1a-491e-9866-5c690b353afe.png)

### Checklist



- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/30290)
